uk.co.compucorp.civicrm.tasksassignments/js/src/crm-tasks-workflows/controllers/casetype.extended.controller.js

```diff
@@ -0,0 +1,52 @@
+/* global angular */
+
+(function (angular) {
+  'use strict';
+
+  angular.module('crm-tasks-workflows.controllers')
+    .controller('CaseTypeExtendedController', CaseTypeExtendedController);
+
+  CaseTypeExtendedController.$inject = [
+    '$controller', '$log', '$scope', 'crmApi', 'apiCalls'
+  ];
+
+  function CaseTypeExtendedController ($controller, $log, $scope, crmApi, apiCalls) {
+    $log.debug('Controller: CaseTypeExtendedController');
+
+    $controller('CaseTypeCtrl', {$scope: $scope, crmApi: crmApi, apiCalls: apiCalls});
+
+    $scope.activityTypeOptions = [];
+    $scope.loading = {
+      activityOptions: false
+    };
+
+    (function init () {
+      fetchActivityTypes();
+    })();
+
+    /**
+     * Fetch activity types which are components of CiviTask or CiviDocument
+     *
+     * @return {Promise}
+     */
+    function fetchActivityTypes () {
+      $scope.loading.activityOptions = true;
+
+      return crmApi('OptionValue', 'get', {
```

Comment:
Now that you can decorate the parameters passed to the router, do you think we can move this one to the route definition?

```js
function routeDecorator ($delegate) {
  $delegate.routes['/caseType/:id'].controller = 'CaseTypeExtendedController';
  $delegate.routes['/caseType/:id'].resolve.activityOptions = function (crmApi) {
    return crmApi('OptionValue', 'get', { /* configuration for CiviTask and CiviDocument activity types */ });
  };

  return $delegate;
}
```

We can also get read of the `.loading.activityOptions` since the screen is not going to be displayed until this request is fetched.